src/client/src/models/Scene.js

```diff
@@ -41,41 +41,66 @@ const Scene = {
     return JSON.parse(localStorage.getItem('scene-list'))
   },
 
+  findSceneIndex: function(sceneid) {
+    // get current list
+    const list = JSON.parse(localStorage.getItem('scene-list'))
+
+    for (var i = 0; i < list.length; i++) {
+      if (sceneid == list[i].id) {
+        return i
+      }
+    }
+
+    return -1
+  },
+
   addScene: function() {
     // get current list
     const list = JSON.parse(localStorage.getItem('scene-list'))
+      // keeps tracks of id numbers in order to not crash into each other
+    var scenes_index = 0
+    // Finds the max scene id in the list
+    if (list.length == 0) {
+      scenes_index = 0
+    } else {
+      scenes_index = 0
+      for (var i = 0; i < list.length; i++) {
+        if (scenes_index < list[i].id) {
+          scenes_index = list[i].id
+        }
+      }
+    }
+    // Adds a new scene
+    scenes_index = scenes_index + 1
 
     // add new scene object
     list.push({
-      id: list.length + 1,
+      id: scenes_index,
+      start: null,
       captions: []
     })
+    // saves the current list
     localStorage.setItem('scene-list', JSON.stringify(list))
   },
 
   deleteScene: function(sceneid) {
-    // decrease sceneid to match with JSON
-    sceneid = sceneid - 1
-
     // get current list
     const list = JSON.parse(localStorage.getItem('scene-list'))
-
-    // goto index at sceneid and take one element out
-    list.splice(sceneid,1)
-
-    // loop and fix id numbers
-    for (var i = 0; i < list.length; i++) {
-      list[i].id = i + 1
+    // loop and delete the scene at that instance in the scene
+    const scene_index = Scene.findSceneIndex(sceneid)
+    if (scene_index != -1) {
+      list.splice(scene_index,1)
     }
+    // saves the current list
```

Comment:
we only delete a scene is we are given a non-negative index, can we save an operation by moving the set scenes to local storage inside the if statement?

Reply:
Change has been made will reflect in the next commit.